addons/voice/org.openhab.voice.googletts/src/main/java/org/openhab/voice/googletts/internal/GoogleCloudAPI.java

```diff
@@ -0,0 +1,318 @@
+
+    /**
+     * Constructor.
+     *
+     * @param cacheFolder Service cache folder
+     */
+    GoogleCloudAPI(File cacheFolder) {
+        this.cacheFolder = cacheFolder;
+    }
+
+    /**
+     * Configuration update.
+     *
+     * @param config New configuration.
+     */
+    void setConfig(GoogleTTSConfig config) {
+        this.config = config;
+        if (checkArch()) {
+            String serviceAccountKey = config.getServiceAccountKey();
+            if (serviceAccountKey != null) {
+                try {
+                    ByteArrayInputStream bis = new ByteArrayInputStream(serviceAccountKey.getBytes());
+                    GoogleCredentials credential = GoogleCredentials.fromStream(bis);
+                    FixedCredentialsProvider credentialProvider = FixedCredentialsProvider.create(credential);
+                    TextToSpeechSettings settings = TextToSpeechSettings.newBuilder().setCredentialsProvider(credentialProvider)
+                            .build();
+                    googleClient = TextToSpeechClient.create(settings);
+                    initialized = true;
+                    initVoices();
+                } catch (IOException e) {
+                    logger.error("Error initializing the service", e);
+                    initialized = false;
+                }
+            } else {
+                googleClient = null;
+                voices.clear();
+            }
+        }
+    }
+
+    private boolean checkArch() {
+        PlatformUtil.Architecture architecture = PlatformUtil.checkArchitecture();
+
+        if (PlatformUtil.Architecture.X86_64 != architecture) {
+            logger.error("The architecture is not x86_64 but {}. Only x86_64 platforms are supported. " , architecture);
+            return false;
+        } else {
+            logger.debug("openHAB is running on architecture {} - supported by Google Cloud TTS API", architecture);
+            return true;
+        }
+    }
+
+    /**
+     * Loads supported audio formats
+     *
+     * @return Set of audio formats
+     */
+    Set<String> getSupportedAudioFormats() {
+        Set<String> formats = new HashSet<>();
+        AudioEncoding[] values = AudioEncoding.values();
+        for (AudioEncoding c : values) {
+            if (c != AudioEncoding.AUDIO_ENCODING_UNSPECIFIED &&
+                    c != AudioEncoding.UNRECOGNIZED) {
+                formats.add(c.toString());
+            }
+        }
+        return formats;
+    }
+
+    /**
+     * Supported locales.
+     *
+     * @return Set of locales
+     */
+    Set<Locale> getSupportedLocales() {
+        return Collections.unmodifiableSet(voices.keySet());
+    }
+
+    /**
+     * Supported voices for locale.
+     *
+     * @param l Locale
+     * @return Set of voices
+     */
+    Set<GoogleTTSVoice> getVoicesForLocale(Locale l) {
+        Set<GoogleTTSVoice> set = voices.get(l);
+        return set != null ? Collections.unmodifiableSet(set) : Collections.EMPTY_SET;
```

Comment:
In Eclipse it shows the warning "Type safety: The expression of type Set needs unchecked conversion to conform to Set<GoogleTTSVoice>" which can be fixed by replacing `Collections.EMPTY_SET` with `Collections.emptySet()`

Reply:
For IntelliJ both are fine. I'll change it to emptySet() call...